app/helpers/injection_helper.rb

```diff
@@ -141,7 +141,7 @@ def inject_rails_flash
   end
 
   def inject_json_ams(name, data, serializer, opts = {})
-    if data.is_a?(Array) || data.class.to_s.end_with?("ActiveRecord_Relation")
+    if data.is_a?(Array) || data.class.to_s.end_with?("ActiveRecord_Relation") || data.class.to_s.end_with?("ActiveRecord_AssociationRelation")
```

Comment:
I wonder if there's a nicer way of handling these cases that doesn't involve explicitly listing multiple class names. Here we're essentially trying to check if the object being passed is singular or plural, right? I'm not sure, but we could possibly use: 

```ruby
if data.respond_to? :each
```

Otherwise, the `Array`, `ActiveRecord::Relation` and `ActiveRecord::AssociationRelation` classes all include the `ActiveModel::ArraySerializerSupport` module, so we could do this:

```ruby
if data.class.include? ActiveModel::ArraySerializerSupport
```

The result will be true for any of the array-like object we are trying to catch in this conditional check.

Reply:
As a side note:
```ruby
object.class.ancestors.include? ActiveRecord::Relation
```
returns `true` for both `ActiveRecord::Relation` and `ActiveRecord::AssociationRelation` objects.

Reply:
this is outdated Matt, I fixed this now by creating two methods, one for lists and one for single objects. Have a look at the latest version.